src/neptune/new/types/type_casting.py

```diff
@@ -0,0 +1,78 @@
+#
+# Copyright (c) 2022, Neptune Labs Sp. z o.o.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import argparse
+from datetime import datetime
+from typing import Any
+
+from neptune.common.deprecation import warn_once
+from neptune.new.handler import Handler
+from neptune.new.internal.utils import (
+    is_bool,
+    is_dict_like,
+    is_float,
+    is_float_like,
+    is_int,
+    is_string,
+    is_string_like,
+)
+from neptune.new.types import (
+    Boolean,
+    File,
+    Integer,
+)
+from neptune.new.types.atoms.datetime import Datetime
+from neptune.new.types.atoms.float import Float
+from neptune.new.types.atoms.string import String
+from neptune.new.types.namespace import Namespace
+from neptune.new.types.value import Value
+from neptune.new.types.value_copy import ValueCopy
+
+
+def cast_value(value: Any) -> Value:
```

Comment:
As we extracted this func (which is so cool!) what do you think about using a single-dispatch? https://docs.python.org/3.7/library/functools.html#functools.singledispatch You should be able to split this function into multiple one that will be triggered base on arguments type.

Reply:
I think it's a good idea.
Let's try it.

Reply:
Or maybe not.
We do casting not only based on type, but also on predicate which looks like it's not supported by singledispatch.